src/gbxml/Test/ReverseTranslator_GTest.cpp

```diff
@@ -268,119 +268,53 @@ TEST_F(gbXMLFixture, ReverseTranslator_FloorSurfaces)
-  ASSERT_TRUE(osurf);
-  EXPECT_EQ("Floor", osurf->surfaceType());
-  space = osurf->space();
-  ASSERT_TRUE(space);
-  EXPECT_EQ("12 Space", space->name().get());
-
-  osurf = model->getModelObjectByName<Surface>("T-8-12-I-F-36 Reversed");
-  ASSERT_TRUE(osurf);
-  EXPECT_EQ("RoofCeiling", osurf->surfaceType());
-  space = osurf->space();
-  ASSERT_TRUE(space);
-  EXPECT_EQ("8 Space", space->name().get());
+  // Check all the surfaces that are supposed to be floors and ceilings
 
+  struct ExpectedSurfaceInfo {
+    ExpectedSurfaceInfo(std::string t_name, std::string t_surfaceType, std::string t_spaceName)
+      : name(t_name), surfaceType(t_surfaceType), spaceName(t_spaceName) {};
+
+    const std::string name;
+    const std::string surfaceType;
+    const std::string spaceName;
+  };
+
+  std::vector<ExpectedSurfaceInfo> expectedSurfaceInfos({
+    ExpectedSurfaceInfo("B-1-U-F-3", "Floor", "1 Space"),
+    ExpectedSurfaceInfo("B-2-U-F-9", "Floor", "2 Space"),
+    ExpectedSurfaceInfo("B-3-U-F-14", "Floor", "3 Space"),
+    ExpectedSurfaceInfo("B-4-U-F-19", "Floor", "4 Space"),
+    ExpectedSurfaceInfo("T-1-5-I-F-6", "Floor", "5 Space"),
+    ExpectedSurfaceInfo("T-1-5-I-F-6 Reversed", "RoofCeiling", "1 Space"),
+    ExpectedSurfaceInfo("T-10-E-R-44", "RoofCeiling", "10 Space"),
+    ExpectedSurfaceInfo("T-11-E-R-48", "RoofCeiling", "11 Space"),
+    ExpectedSurfaceInfo("T-12-E-R-52", "RoofCeiling", "12 Space"),
+    ExpectedSurfaceInfo("T-2-6-I-F-11", "Floor", "6 Space"),
+    ExpectedSurfaceInfo("T-2-6-I-F-11 Reversed", "RoofCeiling", "2 Space"),
+    ExpectedSurfaceInfo("T-3-7-I-F-16", "Floor", "7 Space"),
+    ExpectedSurfaceInfo("T-3-7-I-F-16 Reversed", "RoofCeiling", "3 Space"),
+    ExpectedSurfaceInfo("T-4-8-I-F-20", "Floor", "8 Space"),
+    ExpectedSurfaceInfo("T-4-8-I-F-20 Reversed", "RoofCeiling", "4 Space"),
+    ExpectedSurfaceInfo("T-5-9-I-F-25", "Floor", "9 Space"),
+    ExpectedSurfaceInfo("T-5-9-I-F-25 Reversed", "RoofCeiling", "5 Space"),
+    ExpectedSurfaceInfo("T-6-10-I-F-29", "Floor", "10 Space"),
+    ExpectedSurfaceInfo("T-6-10-I-F-29 Reversed", "RoofCeiling", "6 Space"),
+    ExpectedSurfaceInfo("T-7-11-I-F-33", "Floor", "11 Space"),
+    ExpectedSurfaceInfo("T-7-11-I-F-33 Reversed", "RoofCeiling", "7 Space"),
+    ExpectedSurfaceInfo("T-8-12-I-F-36", "Floor", "12 Space"),
+    ExpectedSurfaceInfo("T-8-12-I-F-36 Reversed", "RoofCeiling", "8 Space"),
+    ExpectedSurfaceInfo("T-9-E-R-39", "RoofCeiling", "9 Space")
+  });
+
+  for (auto& expectedSurfaceInfo : expectedSurfaceInfos) {
+    OptionalSurface _surf = model->getConcreteModelObjectByName<Surface>(expectedSurfaceInfo.name);
+    ASSERT_TRUE(_surf);
+    EXPECT_EQ(expectedSurfaceInfo.surfaceType, _surf->surfaceType()) << "Wrong surfaceType for " << expectedSurfaceInfo.name;
+    auto _space = _surf->space();
+    ASSERT_TRUE(_space);
+    EXPECT_EQ(expectedSurfaceInfo.spaceName, _space->nameString())  << "Wrong space for " << expectedSurfaceInfo.name;
+  }
```

Comment:
Extend existing test and make it dryer. Checking all surfaces makes it fail without fix just like #3951 does.